src/spinner/__tests__/Spinner.test.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import Spinner from '../src/Spinner'
+
+describe('Spinner', () => {
+  jest.useFakeTimers()
+  jest.spyOn(global, 'setTimeout')
+  it('should not crash when rendering', () => {
+    expect(() => {
+      render(<Spinner />)
+    }).not.toThrowError()
+  })
+
+  it('should render', () => {
+    expect(() => {
+      render(<Spinner />)
+    }).not.toBeNull()
```

Comment:
Commenting out the `if` condition in this block `if (!isVisible) {  return null }` in the component does not fail this test. Similar to the below test, I think you'd want something like this to test for its presence:

```tsx
render(<Spinner data-testid="Spinner" />)

expect(screen.getByTestId('Spinner')).not.toBeNull()
```